src/scheduler/server_info.c

```diff
@@ -3247,6 +3264,8 @@ free_status(status *st)
 		free(st->resdef_to_check_rassn);
 	if (st->resdef_to_check_noncons != NULL)
 		free(st->resdef_to_check_noncons);
+	if (st->rel_on_susp != NULL)
+		free(st->rel_on_susp);
```

Comment:
Good catch! I verified the code again, it seems that we don't need to duplicate resdef array but only copy it. I've changed that part of the code.